utils/prepare_for_doc_test.py

```diff
@@ -0,0 +1,94 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Style utils to preprocess files for doc tests."""
+
+import argparse
+import os
+
+
+def maybe_append_new_line(docstring):
+    """
+    Append new line if code snippet is a
+    Python code snippet
+    """
+    lines = docstring.split("\n")
+
+    if lines[0] in ["py", "python"]:
+        # add a "\n" before last line
+        last_line = lines[-1]
+        lines.pop()
+        lines.append("\n" + last_line)
+
+    return "\n".join(lines)
+
+
+def process_doc_file(code_file):
+    """
+    Process given file.
+
+    Args:
+        code_file (`str` or `os.PathLike`): The file in which we want to style the docstring.
+    """
+    with open(code_file, "r", encoding="utf-8", newline="\n") as f:
+        code = f.read()
+
+    # fmt: off
+    splits = code.split("```")
+    splits = [s if i % 2 == 0 else maybe_append_new_line(s) for i, s in enumerate(splits)]
+    clean_code = "```".join(splits)
+    # fmt: on
+
+    diff = clean_code != code
+    if diff:
+        print(f"Overwriting content of {code_file}.")
+        with open(code_file, "w", encoding="utf-8", newline="\n") as f:
```

Comment:
I think we want to put them in some temp dir to run the tests and not overwrite the existing one (otherwise we will get modifications we don't want when running the doctest locally).

Reply:
sounds good!

Reply:
Actually I made two commands now - one that adds the line - one that reverts it. Think this is more intuitive for local testing instead of copying the files and this way we also don't need to add a `temp` path before every file to be tested

Reply:
We should document somewhere the instruction that has to be run before/after testing locally then.

Reply:
Adding a long intro in the beginning of the file. @sgugger - do you think it could be a good idea to add a README.md to `utils` so that community contributors have some docs on how to use the utils scripts?

Reply:
I was thinking more of making a section in the doc page for the tests.

Reply:
Ok that would be good for me as well!